front/app/components/Form/Components/Controls/LinearScaleControl.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+
+// JSON forms
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import { ControlProps } from '@jsonforms/core';
+import { getLabel, sanitizeForClassname } from 'utils/JSONFormUtils';
+
+// Components
+import { Box, Text, Radio, Label } from '@citizenlab/cl2-component-library';
+import { FormLabel } from 'components/UI/FormComponents';
+import VerificationIcon from '../VerificationIcon';
+import ErrorDisplay from '../ErrorDisplay';
+
+const LinearScaleControl = ({
+  data,
+  path,
+  errors,
+  schema,
+  uischema,
+  required,
+  handleChange,
+  id,
+}: ControlProps) => {
+  const maximum = schema?.properties?.rating?.maximum;
+  return (
+    <>
+      <FormLabel
+        htmlFor={sanitizeForClassname(id)}
+        labelValue={getLabel(uischema, schema, path)}
+        optional={!required}
+        subtextValue={uischema.options?.description}
+        subtextSupportsHtml
+      />
+      <Box
+        data-testid="linearScaleControl"
+        display="flex"
+        flexDirection="row"
+        gap="8px"
+        overflow="visible"
+      >
+        <Box alignSelf="flex-end">
+          <Text mr="8px" fontWeight="bold">
+            {uischema.options?.minimum_label}
+          </Text>
+        </Box>
+        <>
+          {[...Array(maximum).keys()].map((i) => {
```

Comment:
Maximum should never be undefined I think, since the default value is 5 (and can only ever be set between values 2 to 7). So even if no maximum is set, the value will be 5 (`schema?.properties?.rating?.maximum` will always have a value).

Reply:
I did not mean `maximum` in `schema?.properties?.rating?.maximum`. I meant the value of the variable `maximum`. It can be undefined because of all the question marks in `schema?.properties?.rating?.maximum`.

Reply:
The question marks are there, but it should never actually return undefined (since the schema will always have a maximum value). We have them there just for typescript. Should I add a " if maximum !== undefined " in the code anyways though for clarity?

Reply:
> We have them there just for typescript
TypeScript should help us, but here it introduces confusion. Do not change the code, because as you wrote the maximum is always there.